pyuvdata/uvcal/initializers.py

```diff
@@ -0,0 +1,533 @@
+
+    Parameters
+    ----------
+    uvdata : UVData
+        UVData object to use as a template.
+    cal_style : str
+        Calibration style. Must be 'sky' or 'redundant'.
+    gain_convention : str
+        Gain convention. Must be 'divide' or 'multiply'.
+    cal_type : str
+        Calibration type. Must be 'gain' or 'delay'. Default 'gain'.
+        Note that in contrast to :func:`new_uvcal`, this function sets the default
+        to 'gain', instead of trying to set it based on the input freq_range.
+    jones_array : array_like of int or str, optional
+        Array of Jones polarizations. Taken from UVData object is possible (i.e. it is
+        single-polarization data), otherwise must be specified. May be specified
+        as 'circular' or 'linear' to indicate all circular or all linear polarizations.
+    wide_band : bool
+        Whether the calibration is wide-band (i.e. one calibration parameter per
+        spectral window instead of per channel). Automatically set to True if cal_type
+        is 'delay', otherwise the default is False. Note that if wide_band is True,
+        the `freq_range` parameter is required.
+    include_uvdata_history : bool
+        Whether to include the UVData object's history in the UVCal object's history.
+        Default is True.
+    spw_array : array_like of int, optional
+        Array of spectral window numbers. This will be taken from the UVData object
+        if at all possible. The only time it is useful to pass this in explicitly is
+        if ``wide_band=True`` and the spectral windows desired are not just
+        ``(0, ..., Nspw-1)``.
+    kwargs : dict
+        All other parameters passed to the :func:`new_uvcal` constructor.
+    """
+    from ..uvdata import UVData
+
+    if not isinstance(uvdata, UVData):
+        raise ValueError("uvdata must be a UVData object.")
+
+    uses_future_shapes = uvdata.future_array_shapes
+    if not uses_future_shapes:
+        uvdata.use_future_array_shapes()
+
+    if cal_type == "delay":
+        wide_band = True
+
+    # If any time-length params are in kwargs, we can't use ANY of them from the UVData
+    if "integration_time" in kwargs or "time_array" in kwargs:
+        integration_time = kwargs.pop("integration_time", None)
+        time_array = kwargs.pop("time_array", None)
+    else:
+        indx = np.unique(uvdata.time_array, return_index=True)[1]
+        integration_time = uvdata.integration_time[indx]
+        time_array = uvdata.time_array[indx]
+
+    # Get frequency-type info from kwargs and uvdata
+    if cal_type == "gain" and not wide_band:
+        if "channel_width" in kwargs or "freq_array" in kwargs:
+            channel_width = kwargs.pop("channel_width", None)
+            freq_array = kwargs.pop("freq_array", None)
+            flex_spw_id_array = kwargs.pop("flex_spw_id_array", None)
+        else:
+            # Just in case the input uvdata is not future_array_shapes compliant, we
+            # flatten
```

Comment:
This confuses me, we're calling `uvdata.use_future_array_shapes()` above.

Reply:
Ah yes, also vestigial. Removed the comment and the `.flatten()` calls.